packages/ui/src/fontManager.ts

```diff
@@ -0,0 +1,253 @@
+import axios from "axios";
+import { getContribLogger, ObjectUtils } from "@playkit-js-contrib/common";
+import ContribFonts = KalturaPlayerTypes.PlayerConfig.ContribFonts;
+import TestingFontOptions = KalturaPlayerTypes.PlayerConfig.TestingFontOptions;
+
+const logger = getContribLogger({
+    module: "ui",
+    class: "FontManager"
+});
+
+const FontKeyPrefix: string = "contrib-plugins-font-";
+
+const DefaultTestingOptions: TestingFontOptions = {
+    text: "abcdefghiiiiiiiiijklmnopqrstuvwwwwwwwwwwxyz0123456789",
+    size: 72,
+    fontName: "monospace"
+};
+
+// shared property for indicating current loaded fontFamily to support multiple contrib managers in a single page
+// ( multiple players instances )
+let currentFontFamily: string = "";
+
+export class FontManager {
+    private static _instance: FontManager;
+
+    private constructor() {}
+
+    public static getInstance(): FontManager {
+        if (!FontManager._instance) {
+            FontManager._instance = new FontManager();
+        }
+        return FontManager._instance;
+    }
+
+    public loadFont(data: ContribFonts): void {
+        // a previous request for loading a font was already made
```

Comment:
see other comments. it is better getting the configuration directly from the core player to prevent errors and perform the default merge in the constructor of this manager 
```
constructor() {
...
const actualConfig = options && options.corePlayer && options.corePlayer.config && options.corePlayer.config.ui ? options.corePlayer.config.ui.fonts : {};
this._config = deepMerge({}, defaultConfig, actualConfig);
....
```